app/views/manageBusinesses/manage/CheckYourAnswers.scala.html

```diff
@@ -0,0 +1,107 @@
+@*
+ * Copyright 2023 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *@
+
+@import views.html.layouts.unifiedLayout
+@import uk.gov.hmrc.govukfrontend.views.html.components._
+@import _root_.implicits.ImplicitDateFormatterImpl
+@import java.time.LocalDate
+@import uk.gov.hmrc.hmrcfrontend.views.html.components.HmrcPageHeading
+@import uk.gov.hmrc.hmrcfrontend.views.viewmodels.pageheading.PageHeading
+@import enums.IncomeSourceJourney._
+@import models.incomeSourceDetails.viewmodels.CheckYourAnswersViewModel
+@import play.twirl.api.HtmlFormat
+
+@this(
+    mainTemplate: unifiedLayout,
+    hmrcPageHeading: HmrcPageHeading,
+    formWithCSRF: FormWithCSRF,
+    govukSummaryList: GovukSummaryList,
+    implicitDateFormatter: ImplicitDateFormatterImpl,
+    govukButton: GovukButton
+)
+
+@(
+    isAgent: Boolean,
+    backUrl: String,
+    viewModel: CheckYourAnswersViewModel,
+    journeyType: String
+)(implicit request: auth.MtdItUser[_], messages: Messages)
+
+@getMessage(key: String, args: String*) = @{
+    messages(s"manageBusinesses.check-answers.$key", args: _*)
+}
+@import implicitDateFormatter.longDate
+
+@getChangeHref = @{
+  val url = journeyType match {
+    case "SE" => controllers.manageBusinesses.manage.routes.ManageIncomeSourceDetailsController.showChange(SelfEmployment, isAgent).url
+    case "UKPROPERTY" => controllers.manageBusinesses.manage.routes.ManageIncomeSourceDetailsController.showChange(UkProperty, isAgent).url
+    case "FOREIGNPROPERTY" => controllers.manageBusinesses.manage.routes.ManageIncomeSourceDetailsController.showChange(ForeignProperty, isAgent).url
+    case _ => ""
+  }
+  url
+}
+
+@getSubmitHref = @{
+    val url = journeyType match {
+        case "SE" => controllers.manageBusinesses.manage.routes.CheckYourAnswersController.submit(isAgent, SelfEmployment)
+        case "UKPROPERTY" => controllers.manageBusinesses.manage.routes.CheckYourAnswersController.submit(isAgent, UkProperty)
+        case "FOREIGNPROPERTY" => controllers.manageBusinesses.manage.routes.CheckYourAnswersController.submit(isAgent, ForeignProperty)
+        case _ => throw new IllegalArgumentException(s"Invalid journeyType: $journeyType")
```

Comment:
we agreed and discuss approach regarding errors in the view;
No exceptions to be thrown in the views